src/placement/jump_map.c

```diff
@@ -307,6 +310,103 @@ get_num_domains(struct pool_domain *curr_dom, enum PL_OP_TYPE op_type)
 
 	return num_dom;
 }
+
+/*
+ * Select the bit position (from the most-significant bit)
+ *
+ * The following 64-bit code selects the position of the rth 1 bit when
+ * counting from the left.
+ *
+ * \param[in]   v       value to find position with
+ * \param[in]   r       bit's desired count [1-64].
+ *
+ * \return      The position of the rth 1 bit in v [0-63]
```

Comment:
Curious what the behavior is when the requested rth bit is higher than total 1 bits. For example, the rth 1 bit is requested, but there are r-1 1 bits? Does it just return 0? 0 is a valid position. Not sure it even matters for the usage of the function.

Reply:
r is always 1 in this usecase. This means give me just first unused bit (first bit set to 1). If all bits are zero it returns 0. Then I check it again if this target is used or not.

Reply:
Simple linear search to process 2^32 searches for 227.46 sec. This function will do the same for 9.41 sec.